server/services/lists/caCaseloadService.ts

```diff
@@ -122,6 +163,37 @@ export default class CaCaseloadService {
     return casesWithComs
   }
 
+  private caCaseload(cases: Case[], searchString: string, view: 'prison' | 'probation'): CaCaseLoad {
+    const caseLoad = this.caseloadViewModel(cases)
+    const showAttentionNeededTab = caseLoad.some(e => e.tabType === CaViewCasesTab.ATTENTION_NEEDED)
+    const searchResults = this.applySearch(searchString, caseLoad)
+    const sortResults = this.applySort(searchResults, view)
+    return { cases: sortResults, showAttentionNeededTab }
+  }
+
+  private caseloadViewModel(cases: Case[]) {
+    return cases.map(c => {
+      const releaseDate = c.licence?.releaseDate || selectReleaseDate(c.nomisRecord)
+      const tabType = determineCaViewCasesTab(c.licence, c.nomisRecord, c.cvlFields)
+      return {
+        kind: c.licence?.kind,
+        licenceId: c.licence?.id,
+        licenceVersionOf: c.licence?.versionOf,
+        name: convertToTitleCase(`${c.nomisRecord.firstName} ${c.nomisRecord.lastName}`.trim()),
+        prisonerNumber: c.nomisRecord.prisonerNumber,
+        probationPractitioner: c.probationPractitioner,
```

Comment:
**question:** Is probationPractitioner always defined? What happens if the case is unallocated?